skfp/distances/rogot_goldberg.py

```diff
@@ -0,0 +1,233 @@
+) -> float:
+    r"""
+    Rogot-Goldberg similarity for vectors of binary values.
+
+    Computes the Rogot-Goldberg similarity [1]_ [2]_ [3]_ for binary data between two
+    input arrays or sparse matrices, using the formula:
+
+    .. math::
+
+        sim(x, y) = \frac{a}{2 * (2a + b + c)} +
+                    \frac{d}{2 * (2d + b + c)}
+
+    where :math:`a`, :math:`b`, :math:`c` and :math:`d` correspond to the number
+    of bit relations between the two vectors:
+
+    - :math:`a` - both are 1 (:math:`|x \cap y|`, common "on" bits)
+    - :math:`b` - :math:`x` is 1, :math:`y` is 0
+    - :math:`c` - :math:`x` is 0, :math:`y` is 1
+    - :math:`d` - both are 0
+
+    Parameters
+    ----------
+    vec_a : {ndarray, sparse matrix}
+        First binary input array or sparse matrix.
+
+    vec_b : {ndarray, sparse matrix}
+        Second binary input array or sparse matrix.
+
+    Returns
+    -------
+    similarity : float
+        Rogot-Goldberg similarity between ``vec_a`` and ``vec_b``.
+
+    References
+    ----------
+    .. [1] `Rogot E., Goldberg I.D.
+        "A proposed index for measuring agreement in test-retest studies."
+        Journal of chronic diseases 19.9 (1966): 991-1006.`
+        <https://doi.org/10.1016/0021-9681(66)90032-4>`_
+
+    .. [2] `Deza M.M., Deza E.
+        "Encyclopedia of Distances."
+        Springer, Berlin, Heidelberg, 2009.
+        <https://doi.org/10.1007/978-3-642-00234-2_1>`_
+
+    .. [3] `RDKit documentation
+        <https://www.rdkit.org/docs/source/rdkit.DataStructs.cDataStructs.html>`_
+
+    Examples
+    --------
+    >>> from skfp.distances import rogot_goldberg_binary_similarity
+    >>> import numpy as np
+    >>> vec_a = np.array([1, 0, 1])
+    >>> vec_b = np.array([1, 0, 1])
+    >>> sim = rogot_goldberg_binary_similarity(vec_a, vec_b)
+    >>> sim
+    1.0
+
+    >>> from scipy.sparse import csr_array
+    >>> vec_a = csr_array([[1, 0, 1]])
+    >>> vec_b = csr_array([[1, 0, 1]])
+    >>> sim = rogot_goldberg_binary_similarity(vec_a, vec_b)
+    >>> sim
+    1.0
+    """
+    _check_finite_values(vec_a)
+    _check_finite_values(vec_b)
+    _check_valid_vectors(vec_a, vec_b)
+
+    if np.sum(vec_a) == 0 == np.sum(vec_b):
+        return 1.0
+
+    if isinstance(vec_a, np.ndarray):
+        vec_a = vec_a.astype(bool)
+        vec_b = vec_b.astype(bool)
+
+        a = np.sum(vec_a & vec_b)
+        b = np.sum(vec_a & ~vec_b)
+        c = np.sum(~vec_a & vec_b)
+        d = np.sum(~vec_a & ~vec_b)
+    else:
+        vec_a = vec_a.tocsr()
+        vec_b = vec_b.tocsr()
+
+        length = vec_a.shape[1]
+        vec_a_idxs = set(vec_a.indices)
+        vec_b_idxs = set(vec_b.indices)
+
+        a = len(vec_a_idxs & vec_b_idxs)
+        b = len(vec_a_idxs - vec_b_idxs)
+        c = len(vec_b_idxs - vec_a_idxs)
+        d = length - (a + b + c)
+
+    first_denom = 2 * a + b + c
+    second_denom = 2 * d + b + c
+
+    if first_denom == 0 or second_denom == 0:
+        return 1.0
```

Comment:
The first denominator will be zero only for all-zero vectors, you have this covered. Only second one can be zero here, in case of all-one vectors. Add comment for that and simplify